File: Proyecto/Sprint.5/sos/src/classes/models/Player.ts
import { Score } from "@/classes/models/Score";

/**
 * @class Player
 * @classdesc Esta clase representa un jugador del juego SOS.
 */
export class Player {
	private readonly name: string;
	private readonly score: Score;

	/**
	 * Crea una instancia de la clase Player.
     * 
	 * @constructor
	 * @param {string} name - El nombre del jugador.
	 */
	constructor(name: string) {
		this.name = name;
		this.score = new Score();
	}

	/**
	 * Obtiene el nombre del jugador.
	 * @returns {string} - El nombre del jugador.
	 */
	public getName(): string {
		return this.name;
	}

	/**
	 * Obtiene el puntaje del jugador.
	 *
	 * @returns {Score} - El puntaje del jugador.
	 */
	public getScore(): Score {
		return this.score;
	}

}

File: Actividades/Semana 8/8.1 ExamenParcial/TicTacToe/src/main/java/com/jhaner/Player.java
package com.jhaner;

/**
 * Representa un jugador en el juego con una letra asignada (X o O).
 */
public class Player {

    private final Letter letter;

    /**
     * {@code @constructor}
     * Crea un nuevo jugador con la letra asignada.
     *
     * @param letter la letra asignada al jugador.
     */
    public Player(Letter letter){
        this.letter = letter;
    }

    /**
     * Devuelve la letra asginada al jugador.
     *
     * @return la letra asignada.
     */
    public Letter getLetter() {
        return letter;
    }
}
